src/stream.rs

```diff
@@ -125,7 +125,7 @@ struct Context {
 }
 
 /// Update the given [State] based on the given [Event].
-async fn handle_event(state: &mut State, event: Event) -> Result<(), Error> {
+async fn handle_event(state: &mut State, event: Event, category: i32) -> Result<(), Error> {
```

Comment:
> If we add this filter, we'd also have to filter the units added in the following line (where it listens for birth events):

Cheers @rkusa for spotting that. Without importing another crate, is the implementation of passing an `i32` instead of `GroupCategory` the best I could do? I tried searching for a few `enum.parse` like functionality in Rust but turning out empty handed.

Reply:
Thanks for taking care of it! You can do `GroupCategory::from_i32(cat)`, it returns a `Option<GroupCategory>`. If you ever want to look at the Rust docs of the code generated from the proto files, you can run the following command (I did the same just now to find this method):


```bash
cargo doc --no-deps -p dcs-grpc-stubs --open
```

Reply:
Ah! Neat trick with generating the documentation with the macros. I was just inferring by convention.